addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/environment-sc-cidr-service.js

```diff
@@ -0,0 +1,221 @@
+    // Before anything, check if the payload is valid
+    this.checkRequest(updateRequest);
+
+    const [environmentScService, lockService] = await this.service(['environmentScService', 'lockService']);
+    const existingEnvironment = await environmentScService.mustFind(requestContext, { id });
+
+    // Check if user is allowed to update cidrs
+    await this.assertAuthorized(
+      requestContext,
+      { action: 'update-sc', conditions: [this._allowAuthorized] },
+      { ...existingEnvironment, updateRequest },
+    );
+
+    await lockService.tryWriteLockAndRun({ id: `${id}-CidrUpdate` }, async () => {
+      // Calculate diff and update CIDR ranges in ingress rules
+      const { currentIngressRules, securityGroupId } = await environmentScService.getSecurityGroupDetails(
+        requestContext,
+        existingEnvironment,
+      );
+
+      // Perform the actual security group ingress rule updates
+      const newCidrList = await this.manageIngressRules(requestContext, {
+        existingEnvironment,
+        securityGroupId,
+        currentIngressRules,
+        updateRequest,
+      });
+
+      // Not storing the changes in the DB, but since all went well
+      // we return the cidr field as part of the env obj
+      existingEnvironment.cidr = newCidrList;
+    });
+
+    // Write audit event
+    await this.audit(requestContext, { action: 'update-environment-sc-cidr', body: existingEnvironment });
+
+    return existingEnvironment;
+  }
+
+  getIpPermission(updateRule, cidrList) {
+    const ipv4List = _.filter(cidrList, cidr => IsCidr.v4(cidr));
+    // const ipv6List = _.filter(cidrList, cidr => IsCidr.v6(cidr));
+
+    const IpPermission = {
+      FromPort: updateRule.fromPort,
+      IpProtocol: updateRule.protocol,
+      IpRanges: _.map(ipv4List, cidr => ({
+        CidrIp: cidr,
+        // For future: Custom description doesn't work, since that has to match exactly what's there in the SG.
+        // This would require storing cidr field in DB
+        // Description: `Updated via SWB`,
+      })),
+      // TODO: Ipv6 calls have to be done separately and cannot habe
+      // Ipv6Ranges: _.map(ipv6List, cidr => ({
+      //   CidrIp: cidr,
+      //   Description: `Updated via SWB`,
+      // })),
+      ToPort: updateRule.toPort,
+    };
+    return IpPermission;
+  }
+
+  // This method is smart enough to determine which IPs to revoke/grant access
+  // by peeking into the ingress rules of the workspace's security group
+  async manageIngressRules(
+    requestContext,
+    { existingEnvironment, securityGroupId, currentIngressRules, updateRequest } = {},
+  ) {
+    const addIpPermissions = [];
+    const revokeIpPermissions = [];
+    const newCidrList = [];
+
+    _.forEach(currentIngressRules, existingRule => {
+      const matchingRule = _.find(updateRequest, {
+        fromPort: existingRule.fromPort,
+        toPort: existingRule.toPort,
+        protocol: existingRule.protocol,
+      });
```

Comment:
Not from the UI. All the CIDR blocks are grouped together in their respective `fromPort`, `toPort`, and `protocol` combinations. 

But since such grouping cannot be guaranteed by someone using the API directly, adding a check makes sense.